pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -127,18 +134,77 @@ func (s *requested) Execute(msg dispatcher.DIDCommMsg) (state, error) {
 			return nil, fmt.Errorf("outbound invitations are not allowed for state %s", s.Name())
 		}
 		// send did-exchange Request
-		return &noOp{}, nil
+		return sendExchangeRequest(msg)
 	case connectionRequest:
 		if msg.Outbound {
 			// send outbound Request
-			return &noOp{}, nil
+			return sendOutboundrequest(msg)
 		}
 		return &responded{}, nil
 	default:
 		return nil, fmt.Errorf("illegal msg type %s for state %s", msg.Type, s.Name())
 	}
 }
 
+func sendExchangeRequest(msg dispatcher.DIDCommMsg) (state, error) {
+	id := uuid.New()
+	//prepare the request
+	request := &Request{
+		Type:  connectionRequest,
+		ID:    id.String(),
+		Label: "Bob", //TODO: How to figure out the label of the request
+	}
+	invitation := &Invitation{}
+	err := json.Unmarshal(msg.Payload, invitation)
+	if err != nil {
+		return &noOp{}, err
+	}
+
+	dest := &Destination{
+		RecipientKeys:   invitation.RecipientKeys,
+		ServiceEndpoint: invitation.ServiceEndpoint,
+		RoutingKeys:     invitation.RoutingKeys,
+	}
+
+	err = msg.Send(request, "", dest.ServiceEndpoint)
+	if err != nil {
+		return &noOp{}, nil
+	}
+
+	return &noOp{}, nil
+}
+
+func sendOutboundrequest(msg dispatcher.DIDCommMsg) (state, error) {
```

Comment:
what's the intended difference between this function and the above?

Reply:
NA refactored